packages/expect/src/utils.ts

```
@@ -16,3 +22,26 @@ export function recordAsyncExpect(test: any, promise: Promise<any> | PromiseLike
 
   return promise
 }
+
+export function wrapSoft(utils: Chai.ChaiUtils, fn: (this: Chai.AssertionStatic & Assertion, ...args: any[]) => void) {
+  return function (this: Chai.AssertionStatic & Assertion, ...args: any[]) {
+    const test: Test = utils.flag(this, 'vitest-test')
+
+    // @ts-expect-error local is untyped
+    const state: MatcherState = test?.context._local
+      ? test.context.expect.getState()
+      : getState((globalThis as any)[GLOBAL_EXPECT])
+
+    if (!test || !state.soft)
```

If soft is true and there is no test, we should throw an error